hw/ip/pattgen/data/pattgen.hjson

```diff
@@ -0,0 +1,160 @@
+        }
+        { bits: "3",
+          resval: "0",
+          name: "POLARITY_CH1",
+	  desc: "Clock (`pcl`) polarity for channel 1.  If high, `pda` signal changes on falling edge of `pcl` signal, otherwise pda changes on rising edge"
+
+          desc: "Enable pattern generator functionality for channel 0"
+        }
+      ]
+    }
+    { name: "PATT_PREDIV_CH0",
+      desc: "PATTGEN pre-divider multi-registers for channel 0",
+      count: "NumRegsPrediv",
+      cname: "PATTGEN",
+      swaccess: "wo",
+      hwaccess: "hro",
+      hwqe: "true",
+      fields: [
+        { bits: "31:0",
+          resval: "0",
+          name: "CLK_RATIO",
+          desc: "Divider ratio from based I/O clock for channel 0"
+        }
+      ]
+    },
+    { name: "PATT_PREDIV_CH1",
+      desc: "PATTGEN pre-divider multi-registers for channel 1",
+      count: "NumRegsPrediv",
+      cname: "PATTGEN",
+      swaccess: "wo",
+      hwaccess: "hro",
+      hwqe: "true",
+      fields: [
+        { bits: "31:0",
+          resval: "0",
+          name: "CLK_RATIO",
+          desc: "Divider ratio from based I/O clock for channel 1"
+        }
+      ]
+    },
+    { multireg: {
+        name: "PATT_DATA_CH0",
+        desc: "PATTGEN Seed pattern multi-registers for channel 0",
+        count: "NumRegsData",
+        cname: "PATTGEN",
+        swaccess: "wo",
+        hwaccess: "hro",
+        hwqe: "true",
+        fields: [
+          { bits: "31:0",
+            resval: "0",
+            name: "DATA",
+            desc: "Seed pattern for channel 0 (2-64 bits)"
+          }
+        ]
+      }
+    },
+    { multireg: {
+        name: "PATT_DATA_CH1",
+        desc: "PATTGEN Seed pattern multi-registers for channel 1",
+        count: "NumRegsData",
+        cname: "PATTGEN",
+        swaccess: "wo",
+        hwaccess: "hro",
+        hwqe: "true",
+        fields: [
+          { bits: "31:0",
+            resval: "0",
+            name: "DATA",
+            desc: "Seed pattern for channel 1 (2-64 bits)"
+          }
+        ]
+      }
+    },
+    { name: "PATT_SIZE",
+      desc: "PATTGEN pattern length",
+      swaccess: "wo",
+      hwaccess: "hro",
+      hwqe: "true",
+      fields: [
+        { bits: "5:0",
+          resval: "0",
+          name: "LEN_CH0",
+          desc: "Length of the seed pattern for channel 0 (1-64 bits), minus 1"
+        }
+        { bits: "15:6",
+          resval: "0",
+          name: "REPS_CH0",
+          desc: "Number of pattern repetitions for channel 0, minus 1"
+        }	  	
+        { bits: "21:16",
+          resval: "0",
+          name: "LEN_CH1",
+          desc: "Length of the seed pattern for channel 1 (1-64 bits), minus 1"
+        }
+        { bits: "31:22",
+          resval: "0",
+          name: "REPS_CH1",
+          desc: "Number of pattern repetitions for channel 1, minus 1"
+        }	  	
```

Comment:
Please remove all trailing whitespace from this file (there's a tab here and a couple more above).